Formal2024/Exercises/ex5.lean

```diff
@@ -53,21 +59,32 @@ theorem Nat.exists_unique_le : ∃! n m : Nat, m ≤ n := sorry
 theorem existence_implies_everything_iff {α : Type*} (r : α → Prop) :
   (∃ x, (r x → ∀ y, r y)) ↔ Nonempty α := sorry
 
+section Part_4
 /-! # Part 4 : non-equal types
 
-  we will prove the following lemmas and theorems about non-equal types.
+Firstly, some definitions:
+* `Cardinal` is a structure that represents the cardinality of a type.
+  We can use it to compare the cardinality of two types.
+  In order to use the notation `#α`, which exactly means "the cardinality of `α`", we need to `open Cardinal` first.
+* `Countable α` is defined as `Prop`, which is determined by whether there exists an injective map `α → ℕ`.
+
 
-  [-] `ne_types_of_ne_cardinal`: we have non-equal types given their cardinality is non-equal
+We will prove the following lemmas and theorems about non-equal types.
 
-  [-] `ne_types_of_countable_uncountable`: if a type is countable and the other isn't, then they are non-equal
+* `ne_types_of_ne_cardinal`: we have non-equal types given their cardinality is non-equal
 
-  [-] `ne_types_of_different_property`: generalising the above two,
+* `ne_types_of_countable_uncountable`: if a type is countable and the other isn't, then they are non-equal
```

Comment:
Shouldn't this one go first because it's a special case of the other?

Reply:
eh, they're equivalent cases, it doesn't matter

Reply:
Not that it matters too much, but the first one is more general, cause it holds also for finite sets or higher cardinals than aleph_1